phabricator.js

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+async function injectButton(block) {
+  let path = block.querySelector('h1.differential-file-icon-header').textContent;
+  let buttonDiv = block.querySelector('div.differential-changeset-buttons');
+
+  function getLines() {
+    return block.querySelectorAll('table.differential-diff tbody tr th:first-child');
+  }
+
+  let button = document.createElement('button');
+  button.type = 'button';
+  button.textContent = 'Loading Code Coverage...';
+  button.disabled = true;
+  button.className = 'button button-grey';
+  buttonDiv.append(button);
+
+  let parentRevision = await parentRevisionPromise;
+  let data;
+  if (!parentRevision) {
+    data = null;
+  } else {
+    ({ data } = await fetchCoverage(parentRevision, path));
```

Comment:
You'll need to handle the error here, if data has an `error` key.

Reply:
The only check of this sort I see in the rest of the code is this: https://github.com/mozilla/code-coverage-addon/blob/28b045b795cd85dceabcfc39e51e0a6d618125c0/mozreview.js#L49

I've done something similar, also checking for a `error` key.

Reply:
Here are two cases where we check for the error:
https://github.com/mozilla/code-coverage-addon/blob/d898be400d0554adb4c9f01ad58e8b3054be7035/socorro.js#L61
https://github.com/mozilla/code-coverage-addon/blob/d898be400d0554adb4c9f01ad58e8b3054be7035/bugzilla.js#L78

We're not very consistent in error handling though, we should make it better.